comp/core/workloadmeta/impl/store_test.go

```diff
@@ -31,19 +31,28 @@ const (
 	barSource       = "bar"
 )
 
-func newWorkloadmetaObject(deps dependencies) *workloadmeta {
-	return NewWorkloadMeta(deps).Comp.(*workloadmeta)
+type testDependencies struct {
+	fx.In
+	Config config.Component
 }
 
-func TestHandleEvents(t *testing.T) {
-
-	deps := fxutil.Test[dependencies](t, fx.Options(
-		fx.Provide(func() log.Component { return logmock.New(t) }),
+func newWorkloadmetaObject(t *testing.T) *workloadmeta {
+	testDeps := fxutil.Test[testDependencies](t, fx.Options(
```

Comment:
As you are still use `fxutil.Test`, `fx` is still used in `impl` folder. 
I am not sure removing `fxutil.Test` is a great idea for now, so you can probably keep it.

Reply:
Indeed. I noticed it but thought it was fine in test files.
Any suggestion about how to remove this `fx` dependency here? It's only used in the line that you mentioned and the `testDependencies` struct defined above.

Reply:
I think you can keep your code as it is.